build.gradle

```diff
@@ -46,6 +46,8 @@ allprojects {
   description = "Elasticsearch subproject ${project.path}"
 }
 
+BuildPlugin.configureRepositories(project)
```

Comment:
This is kinda goofy. Why not instead rip `configureRepositories()` out into it's on `RepositoriesPlugin` which `BuildPlugin` applies and then we can apply it by itself to the root project.

Reply:
I agree it would be better, just to keep the focus of this PR. 
There are multiple method like this one, and there's cleanup that we should do around pieces of build plugin going into others in general ( probably by having a base plugin ) that I didn't want to get into now.